### PR TITLE
feat: add safe-mode and tx-pause pallets

### LINKED ISSUE #72: Safe Mode capability
Having the ability to temporarily pause transfers & normal operations on a live parachain is a valuable mechanism, when an issue / attack occurs. This is especially useful during the launch period of a parachain whereby the runtime code might not always have been battle-tested yet. Past this initial period, the chain’s Governance body (or Sudo account(s)) could decide to remove this capability to fully decentralize the network. The Substrate FRAME library contains the [SafeMode & TxPause pallets](https://github.com/paritytech/substrate/pull/12092) which allow to implement such a mechanism.

- [ ] Integrate the FRAME SafeMode & TxPause pallets with sane defaults in both devnet & mainnet runtimes (disabled by default in both runtimes).
- [ ] SafeMode should only allow balance transfers as whitelisted call. Permissionless operations (enter, extend) should be disabled. Only a privileged origin can enable / disable safe mode: `ForceEnterOrigin`, `ForceExtendOrigin` & `ForceExitOrigin` to be configured as the Root origin (Sudo).
- [ ] TxPause `PauseOrigin` & `UnpauseOrigin` to be configured as the Root origin (Sudo).

Depends on #71

### PR DESCRIPTION
**Pull request by Moliholy**
Closes #72

### CODE REVIEW

File: runtime/common/src/lib.rs [hunk 1]
```diff
@@ -30,6 +30,13 @@ pub const MINUTES: BlockNumber = 60_000 / (MILLISECS_PER_BLOCK as BlockNumber);
 pub const HOURS: BlockNumber = MINUTES * 60;
 pub const DAYS: BlockNumber = HOURS * 24;
 
+pub const UNITS: Balance = 1_000_000_000_000;
+pub const QUID: Balance = UNITS / 30;
+pub const CENTS: Balance = QUID / 100;
+pub const GRAND: Balance = QUID * 1_000;
+pub const MILLICENTS: Balance = CENTS / 1_000;
+pub const DOLLARS: Balance = CENTS / 1_000;
```
Comment (weezy20) [inline runtime/common/src/lib.rs:38]: We are actually sticking to standard units (no quids, cents, dollars) for the parachain templates. We used to have them early on but then decided against having biased names.  Would suggest these values be recalculated in places used in terms of `UNITS` or `MILLIUNITS`

File: runtime/devnet/src/lib.rs [hunk 2]
```diff
@@ -611,6 +614,67 @@ impl pallet_collator_selection::Config for Runtime {
...
+impl pallet_safe_mode::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	type WhitelistedCalls = SafeModeWhitelistedCalls;
+	type EnterDuration = EnterDuration;
+	type ExtendDuration = ExtendDuration;
+	type EnterDepositAmount = EnterDepositAmount;
+	type ExtendDepositAmount = ExtendDepositAmount;
+	type ForceEnterOrigin = EnsureRootWithSuccess<AccountId, ConstU32<9>>;
```
[the remark this answers is not in the document]
Comment (Moliholy) [reply runtime/devnet/src/lib.rs:649]: Addressed in [1641d5e](https://github.com/paritytech/extended-parachain-template/pull/94/commits/1641d5e17f215892214747bfd4cee2766ab4df45).

File: runtime/common/src/lib.rs (hunk 1 above)
Comment (Moliholy) [reply runtime/common/src/lib.rs]: Addressed in [9f19c32](https://github.com/paritytech/extended-parachain-template/pull/94/commits/9f19c32601f15acb7cff649cffa95da7c54c90a2).

Please take a look. I set the `enter` operation to 1 DOT and the extend operation to 0.5 DOT, but those values might not necessarily be good enough.

File: runtime/devnet/src/lib.rs [hunk 3]
```diff
@@ -630,9 +630,9 @@ impl Contains<RuntimeCall> for SafeModeWhitelistedCalls {
 
 parameter_types! {
 	pub const EnterDuration: BlockNumber = 4 * HOURS;
-	pub const EnterDepositAmount: Balance = UNITS;
+	pub const EnterDepositAmount: Option<Balance> = None;
```
Comment (weezy20) [inline runtime/devnet/src/lib.rs:633]: Any reason for preferring `Option::None` over `Balance::zero()`?

Comment (Moliholy) [reply runtime/devnet/src/lib.rs:633]: According to [the docs](https://github.com/paritytech/polkadot-sdk/blob/release-polkadot-v1.1.0/substrate/frame/safe-mode/src/lib.rs#L89) `None` is a sane default, while `Balance::zero()` is not specified as such and I suspect it'd allow anyone to enter into safe mode at zero cost.

Comment (weezy20) [reply runtime/devnet/src/lib.rs:633]: I see! thanks for clarifying

**Review by weezy20 — DISMISSED** [review]

File: runtime/common/src/lib.rs [hunk 4]
```diff
@@ -30,6 +30,8 @@ pub const MINUTES: BlockNumber = 60_000 / (MILLISECS_PER_BLOCK as BlockNumber);
 pub const HOURS: BlockNumber = MINUTES * 60;
 pub const DAYS: BlockNumber = HOURS * 24;
 
+pub const UNITS: Balance = 10_000_000_000;
```
Comment (stiiifff) [inline runtime/common/src/lib.rs:33]: Is this used somewhere ?

Comment (Moliholy) [reply runtime/common/src/lib.rs]: No, it was used before but forgot to remove it. Thanks!

Addressed in [bea00b4](https://github.com/paritytech/extended-parachain-template/pull/94/commits/bea00b40c158499dc1c86c2bbcae517406782b5c).

**Comment by stiiifff** [comment]
Awesome, another nice addition to the EPT template. LGTM !

**Review by stiiifff — APPROVED** [review]

**Pull request merged**